```
Error running mint automation
+ mkdir -p 8397-b912bc6/gopath/src/github.com/minio
+ git -C 8397-b912bc6/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 8397-b912bc6/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 8397-b912bc6/gopath/src/github.com/minio/minio fetch --quiet minio pull/8397/head:pr8397
+ git -C 8397-b912bc6/gopath/src/github.com/minio/minio checkout --quiet pr8397
+ GO111MODULE=on
+ GOPATH=/mnt/nvme/mint-auto/8397-b912bc6/gopath
+ make -C 8397-b912bc6/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading google.golang.org/api v0.4.0
go: downloading github.com/minio/hdfs/v3 v3.0.1
go: downloading github.com/Azure/azure-sdk-for-go v33.4.0+incompatible
go: downloading github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading github.com/klauspost/compress v1.8.3
go: extracting github.com/dustin/go-humanize v1.0.0
go: downloading github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: extracting github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/minio/sha256-simd v0.1.1
go: extracting github.com/minio/hdfs/v3 v3.0.1
go: extracting github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: extracting github.com/minio/sha256-simd v0.1.1
go: downloading github.com/minio/mc v0.0.0-20190924013003-643835013047
go: downloading github.com/gorilla/rpc v1.2.0
go: downloading github.com/minio/cli v1.21.0
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: extracting github.com/gorilla/rpc v1.2.0
go: downloading github.com/cheggaaa/pb v1.0.28
go: downloading github.com/minio/minio-go/v6 v6.0.39
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/cheggaaa/pb v1.0.28
go: extracting github.com/minio/cli v1.21.0
go: downloading github.com/rs/cors v1.6.0
go: downloading github.com/miekg/dns v1.1.8
go: extracting github.com/minio/mc v0.0.0-20190924013003-643835013047
go: extracting github.com/rs/cors v1.6.0
go: downloading github.com/ncw/directio v1.0.5
go: extracting github.com/ncw/directio v1.0.5
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/minio/minio-go/v6 v6.0.39
go: extracting github.com/miekg/dns v1.1.8
go: downloading gopkg.in/ldap.v3 v3.0.3
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: extracting gopkg.in/ldap.v3 v3.0.3
go: downloading go.uber.org/atomic v1.3.2
go: downloading github.com/gogo/protobuf v1.2.1
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: extracting go.uber.org/atomic v1.3.2
go: downloading github.com/minio/gokrb5/v7 v7.2.5
go: extracting github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/minio/lsync v1.0.1
go: extracting github.com/minio/lsync v1.0.1
go: downloading github.com/mattn/go-isatty v0.0.7
go: extracting github.com/mattn/go-isatty v0.0.7
go: downloading github.com/Azure/go-autorest v11.7.0+incompatible
go: extracting github.com/coreos/etcd v3.3.12+incompatible
go: downloading github.com/minio/sio v0.2.0
go: extracting github.com/Azure/go-autorest v11.7.0+incompatible
go: downloading golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: extracting github.com/minio/sio v0.2.0
go: downloading github.com/klauspost/readahead v1.3.0
go: extracting github.com/gogo/protobuf v1.2.1
go: downloading github.com/nats-io/stan.go v0.4.5
go: extracting github.com/klauspost/readahead v1.3.0
go: downloading github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: extracting golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: extracting github.com/nats-io/stan.go v0.4.5
go: downloading github.com/mattn/go-colorable v0.1.1
go: extracting github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: downloading gopkg.in/Shopify/sarama.v1 v1.20.0
go: extracting github.com/mattn/go-colorable v0.1.1
go: downloading github.com/prometheus/common v0.2.0
go: extracting gopkg.in/Shopify/sarama.v1 v1.20.0
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting github.com/prometheus/common v0.2.0
go: downloading github.com/fatih/color v1.7.0
go: extracting github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: extracting github.com/fatih/color v1.7.0
go: downloading golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: extracting github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: extracting github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/gorilla/mux v1.7.0
go: extracting github.com/gorilla/mux v1.7.0
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading github.com/nats-io/nuid v1.0.1
go: extracting github.com/nats-io/nuid v1.0.1
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/eapache/queue v1.1.0
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/djherbis/atime v1.0.0
go: extracting github.com/eapache/queue v1.1.0
go: extracting github.com/djherbis/atime v1.0.0
go: extracting github.com/Azure/azure-sdk-for-go v33.4.0+incompatible
go: extracting github.com/klauspost/compress v1.8.3
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/pkg/errors v0.8.1
go: extracting github.com/pkg/errors v0.8.1
go: downloading github.com/davecgh/go-spew v1.1.1
go: extracting github.com/davecgh/go-spew v1.1.1
go: downloading github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: extracting github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: extracting github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading github.com/klauspost/cpuid v1.2.1
go: extracting github.com/klauspost/cpuid v1.2.1
go: downloading github.com/shirou/gopsutil v2.18.12+incompatible
go: extracting github.com/shirou/gopsutil v2.18.12+incompatible
go: downloading github.com/pkg/profile v1.3.0
go: extracting github.com/pkg/profile v1.3.0
go: downloading github.com/hashicorp/vault v1.1.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: extracting contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: downloading github.com/satori/go.uuid v1.2.0
go: extracting github.com/satori/go.uuid v1.2.0
go: downloading github.com/pierrec/lz4 v2.0.5+incompatible
go: extracting github.com/minio/gokrb5/v7 v7.2.5
go: downloading go.opencensus.io v0.21.0
go: downloading github.com/minio/dsync v1.0.0
go: downloading github.com/klauspost/reedsolomon v1.9.3
go: extracting github.com/minio/dsync v1.0.0
go: downloading github.com/minio/dsync/v2 v2.0.0
go: extracting github.com/klauspost/reedsolomon v1.9.3
go: extracting github.com/pierrec/lz4 v2.0.5+incompatible
go: extracting github.com/minio/dsync/v2 v2.0.0
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: extracting go.opencensus.io v0.21.0
go: extracting google.golang.org/api v0.4.0
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading google.golang.org/grpc v1.20.1
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: downloading github.com/census-instrumentation/opencensus-proto v0.2.0
go: downloading github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: downloading github.com/eapache/go-resiliency v1.2.0
go: extracting github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: extracting github.com/coredns/coredns v1.4.0
go: extracting github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: extracting github.com/census-instrumentation/opencensus-proto v0.2.0
go: downloading github.com/alecthomas/participle v0.2.1
go: extracting github.com/go-sql-driver/mysql v1.4.1
go: downloading github.com/golang/snappy v0.0.1
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/lib/pq v1.0.0
go: extracting github.com/alecthomas/participle v0.2.1
go: downloading github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: extracting github.com/golang/snappy v0.0.1
go: extracting github.com/lib/pq v1.0.0
go: downloading github.com/nats-io/nats.go v1.8.0
go: extracting github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: extracting google.golang.org/grpc v1.20.1
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/nats-io/nats.go v1.8.0
go: extracting github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: downloading github.com/json-iterator/go v1.1.7
go: downloading golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: downloading github.com/gorilla/handlers v1.4.0
go: downloading github.com/golang/protobuf v1.3.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting github.com/gorilla/handlers v1.4.0
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: extracting github.com/json-iterator/go v1.1.7
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/klauspost/pgzip v1.2.1
go: extracting gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading cloud.google.com/go v0.37.2
go: extracting github.com/golang/protobuf v1.3.1
go: downloading github.com/fatih/structs v1.1.0
go: downloading gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/nats-io/nkeys v0.0.2
go: downloading gopkg.in/ini.v1 v1.48.0
go: downloading github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting github.com/klauspost/pgzip v1.2.1
go: downloading github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting gopkg.in/ini.v1 v1.48.0
go: extracting github.com/nats-io/nkeys v0.0.2
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: extracting github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading github.com/rjeczalik/notify v0.9.2
go: extracting github.com/fatih/structs v1.1.0
go: extracting gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/tidwall/gjson v1.2.1
go: extracting github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/modern-go/reflect2 v1.0.1
go: extracting github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: extracting github.com/rjeczalik/notify v0.9.2
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/nsqio/go-nsq v1.0.7
go: extracting github.com/modern-go/reflect2 v1.0.1
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading golang.org/x/text v0.3.2
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/tidwall/gjson v1.2.1
go: downloading github.com/minio/highwayhash v1.0.0
go: extracting github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/tidwall/match v1.0.1
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting github.com/minio/highwayhash v1.0.0
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/tidwall/pretty v1.0.0
go: extracting github.com/tidwall/match v1.0.1
go: downloading golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: extracting github.com/tidwall/pretty v1.0.0
go: downloading git.apache.org/thrift.git v0.12.0
go: extracting golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: extracting github.com/hashicorp/vault v1.1.0
go: extracting git.apache.org/thrift.git v0.12.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: extracting github.com/hashicorp/hcl v1.0.0
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/hashicorp/go-retryablehttp v0.5.3
go: extracting github.com/hashicorp/go-retryablehttp v0.5.3
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: extracting github.com/hashicorp/errwrap v1.0.0
go: extracting golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: extracting cloud.google.com/go v0.37.2
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/ryanuber/go-glob v1.0.0
go: extracting golang.org/x/text v0.3.2
go: extracting google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: downloading github.com/googleapis/gax-go v2.0.0+incompatible
go: extracting github.com/googleapis/gax-go v2.0.0+incompatible
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: finding google.golang.org/api v0.4.0
go: finding github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/minio/minio-go/v6 v6.0.39
go: finding github.com/minio/sha256-simd v0.1.1
go: finding github.com/minio/cli v1.21.0
go: finding github.com/Azure/go-autorest v11.7.0+incompatible
go: finding cloud.google.com/go v0.37.2
go: finding github.com/minio/gokrb5/v7 v7.2.5
go: finding github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: finding github.com/json-iterator/go v1.1.7
go: finding github.com/minio/mc v0.0.0-20190924013003-643835013047
go: finding github.com/Azure/azure-sdk-for-go v33.4.0+incompatible
go: finding github.com/gorilla/mux v1.7.0
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding github.com/minio/hdfs/v3 v3.0.1
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding github.com/minio/highwayhash v1.0.0
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding github.com/fatih/color v1.7.0
go: finding github.com/mattn/go-colorable v0.1.1
go: finding golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding github.com/mattn/go-isatty v0.0.7
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: finding google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: finding google.golang.org/grpc v1.20.1
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding gopkg.in/ini.v1 v1.48.0
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding go.opencensus.io v0.21.0
go: finding github.com/rjeczalik/notify v0.9.2
go: finding contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: finding github.com/golang/protobuf v1.3.1
go: finding github.com/satori/go.uuid v1.2.0
go: finding golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: finding github.com/gogo/protobuf v1.2.1
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding github.com/census-instrumentation/opencensus-proto v0.2.0
go: finding golang.org/x/text v0.3.2
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/mattn/go-runewidth v0.0.4
go: finding github.com/coreos/etcd v3.3.12+incompatible
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding github.com/gorilla/handlers v1.4.0
go: finding github.com/gorilla/rpc v1.2.0
go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: finding github.com/klauspost/compress v1.8.3
go: finding github.com/klauspost/cpuid v1.2.1
go: finding github.com/klauspost/readahead v1.3.0
go: finding github.com/klauspost/reedsolomon v1.9.3
go: finding github.com/miekg/dns v1.1.8
go: finding github.com/minio/dsync/v2 v2.0.0
go: finding github.com/minio/lsync v1.0.1
go: finding github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: finding gopkg.in/ldap.v3 v3.0.3
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: finding github.com/eclipse/paho.mqtt.golang v1.2.0
go: finding github.com/go-sql-driver/mysql v1.4.1
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/lib/pq v1.0.0
go: finding github.com/nats-io/nats.go v1.8.0
go: finding github.com/nats-io/nkeys v0.0.2
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/nats-io/stan.go v0.4.5
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: finding github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: finding gopkg.in/Shopify/sarama.v1 v1.20.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/eapache/go-resiliency v1.2.0
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding github.com/eapache/queue v1.1.0
go: finding github.com/pierrec/lz4 v2.0.5+incompatible
go: finding github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: finding github.com/hashicorp/vault v1.1.0
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/hashicorp/go-retryablehttp v0.5.3
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding github.com/ncw/directio v1.0.5
go: finding github.com/minio/sio v0.2.0
go: finding github.com/coredns/coredns v1.4.0
go: finding github.com/fatih/structs v1.1.0
go: finding github.com/tidwall/gjson v1.2.1
go: finding github.com/tidwall/match v1.0.1
go: finding github.com/tidwall/pretty v1.0.0
go: finding gopkg.in/yaml.v2 v2.2.2
go: finding github.com/shirou/gopsutil v2.18.12+incompatible
go: finding github.com/klauspost/pgzip v1.2.1
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding github.com/alecthomas/participle v0.2.1
go: finding github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: finding git.apache.org/thrift.git v0.12.0
go: finding github.com/pkg/profile v1.3.0
go: finding github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: finding github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: finding github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: finding github.com/prometheus/common v0.2.0
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: finding github.com/rs/cors v1.6.0
go: finding go.uber.org/atomic v1.3.2
# github.com/minio/minio/pkg/event/target
pkg/event/target/mqtt.go:202:76: undefined: err
make: *** [build] Error 2

```

--- a/CREDITS
+++ b/CREDITS
@@ -3303,6 +3303,98 @@ Everyone is permitted to copy and distribute copies of this Agreement, but in or
 This Agreement is governed by the laws of the State of New York and the intellectual property laws of the United States of America. No party to this Agreement will bring a legal action under this Agreement more than one year after the cause of action arose. Each party waives its rights to a jury trial in any resulting litigation.
 ================================================================
 
+github.com/eclipse/paho.mqtt.golang
+https://github.com/eclipse/paho.mqtt.golang
+----------------------------------------------------------------
+Eclipse Public License - v 1.0
+
+THE ACCOMPANYING PROGRAM IS PROVIDED UNDER THE TERMS OF THIS ECLIPSE PUBLIC LICENSE ("AGREEMENT"). ANY USE, REPRODUCTION OR DISTRIBUTION OF THE PROGRAM CONSTITUTES RECIPIENT'S ACCEPTANCE OF THIS AGREEMENT.
+
+1. DEFINITIONS
+
+"Contribution" means:
+
+a) in the case of the initial Contributor, the initial code and documentation distributed under this Agreement, and
+
+b) in the case of each subsequent Contributor:
+
+i) changes to the Program, and
+
+ii) additions to the Program;
+
+where such changes and/or additions to the Program originate from and are distributed by that particular Contributor. A Contribution 'originates' from a Contributor if it was added to the Program by such Contributor itself or anyone acting on such Contributor's behalf. Contributions do not include additions to the Program which: (i) are separate modules of software distributed in conjunction with the Program under their own license agreement, and (ii) are not derivative works of the Program.
+
+"Contributor" means any person or entity that distributes the Program.
+
+"Licensed Patents" mean patent claims licensable by a Contributor which are necessarily infringed by the use or sale of its Contribution alone or when combined with the Program.
+
+"Program" means the Contributions distributed in accordance with this Agreement.
+
+"Recipient" means anyone who receives the Program under this Agreement, including all Contributors.
+
+2. GRANT OF RIGHTS
+
+a) Subject to the terms of this Agreement, each Contributor hereby grants Recipient a non-exclusive, worldwide, royalty-free copyright license to reproduce, prepare derivative works of, publicly display, publicly perform, distribute and sublicense the Contribution of such Contributor, if any, and such derivative works, in source code and object code form.
+
+b) Subject to the terms of this Agreement, each Contributor hereby grants Recipient a non-exclusive, worldwide, royalty-free patent license under Licensed Patents to make, use, sell, offer to sell, import and otherwise transfer the Contribution of such Contributor, if any, in source code and object code form. This patent license shall apply to the combination of the Contribution and the Program if, at the time the Contribution is added by the Contributor, such addition of the Contribution causes such combination to be covered by the Licensed Patents. The patent license shall not apply to any other combinations which include the Contribution. No hardware per se is licensed hereunder.
+
+c) Recipient understands that although each Contributor grants the licenses to its Contributions set forth herein, no assurances are provided by any Contributor that the Program does not infringe the patent or other intellectual property rights of any other entity. Each Contributor disclaims any liability to Recipient for claims brought by any other entity based on infringement of intellectual property rights or otherwise. As a condition to exercising the rights and licenses granted hereunder, each Recipient hereby assumes sole responsibility to secure any other intellectual property rights needed, if any. For example, if a third party patent license is required to allow Recipient to distribute the Program, it is Recipient's responsibility to acquire that license before distributing the Program.
+
+d) Each Contributor represents that to its knowledge it has sufficient copyright rights in its Contribution, if any, to grant the copyright license set forth in this Agreement.
+
+3. REQUIREMENTS
+
+A Contributor may choose to distribute the Program in object code form under its own license agreement, provided that:
+
+a) it complies with the terms and conditions of this Agreement; and
+
+b) its license agreement:
+
+i) effectively disclaims on behalf of all Contributors all warranties and conditions, express and implied, including warranties or conditions of title and non-infringement, and implied warranties or conditions of merchantability and fitness for a particular purpose;
+
+ii) effectively excludes on behalf of all Contributors all liability for damages, including direct, indirect, special, incidental and consequential damages, such as lost profits;
+
+iii) states that any provisions which differ from this Agreement are offered by that Contributor alone and not by any other party; and
+
+iv) states that source code for the Program is available from such Contributor, and informs licensees how to obtain it in a reasonable manner on or through a medium customarily used for software exchange.
+
+When the Program is made available in source code form:
+
+a) it must be made available under this Agreement; and
+
+b) a copy of this Agreement must be included with each copy of the Program.
+
+Contributors may not remove or alter any copyright notices contained within the Program.
+
+Each Contributor must identify itself as the originator of its Contribution, if any, in a manner that reasonably allows subsequent Recipients to identify the originator of the Contribution.
+
+4. COMMERCIAL DISTRIBUTION
+
+Commercial distributors of software may accept certain responsibilities with respect to end users, business partners and the like. While this license is intended to facilitate the commercial use of the Program, the Contributor who includes the Program in a commercial product offering should do so in a manner which does not create potential liability for other Contributors. Therefore, if a Contributor includes the Program in a commercial product offering, such Contributor ("Commercial Contributor") hereby agrees to defend and indemnify every other Contributor ("Indemnified Contributor") against any losses, damages and costs (collectively "Losses") arising from claims, lawsuits and other legal actions brought by a third party against the Indemnified Contributor to the extent caused by the acts or omissions of such Commercial Contributor in connection with its distribution of the Program in a commercial product offering. The obligations in this section do not apply to any claims or Losses relating to any actual or alleged intellectual property infringement. In order to qualify, an Indemnified Contributor must: a) promptly notify the Commercial Contributor in writing of such claim, and b) allow the Commercial Contributor to control, and cooperate with the Commercial Contributor in, the defense and any related settlement negotiations. The Indemnified Contributor may participate in any such claim at its own expense.
+
+For example, a Contributor might include the Program in a commercial product offering, Product X. That Contributor is then a Commercial Contributor. If that Commercial Contributor then makes performance claims, or offers warranties related to Product X, those performance claims and warranties are such Commercial Contributor's responsibility alone. Under this section, the Commercial Contributor would have to defend claims against the other Contributors related to those performance claims and warranties, and if a court requires any other Contributor to pay any damages as a result, the Commercial Contributor must pay those damages.
+
+5. NO WARRANTY
+
+EXCEPT AS EXPRESSLY SET FORTH IN THIS AGREEMENT, THE PROGRAM IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED INCLUDING, WITHOUT LIMITATION, ANY WARRANTIES OR CONDITIONS OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Each Recipient is solely responsible for determining the appropriateness of using and distributing the Program and assumes all risks associated with its exercise of rights under this Agreement , including but not limited to the risks and costs of program errors, compliance with applicable laws, damage to or loss of data, programs or equipment, and unavailability or interruption of operations.
+
+6. DISCLAIMER OF LIABILITY
+
+EXCEPT AS EXPRESSLY SET FORTH IN THIS AGREEMENT, NEITHER RECIPIENT NOR ANY CONTRIBUTORS SHALL HAVE ANY LIABILITY FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING WITHOUT LIMITATION LOST PROFITS), HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OR DISTRIBUTION OF THE PROGRAM OR THE EXERCISE OF ANY RIGHTS GRANTED HEREUNDER, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
+
+7. GENERAL
+
+If any provision of this Agreement is invalid or unenforceable under applicable law, it shall not affect the validity or enforceability of the remainder of the terms of this Agreement, and without further action by the parties hereto, such provision shall be reformed to the minimum extent necessary to make such provision valid and enforceable.
+
+If Recipient institutes patent litigation against any entity (including a cross-claim or counterclaim in a lawsuit) alleging that the Program itself (excluding combinations of the Program with other software or hardware) infringes such Recipient's patent(s), then such Recipient's rights granted under Section 2(b) shall terminate as of the date such litigation is filed.
+
+All Recipient's rights under this Agreement shall terminate if it fails to comply with any of the material terms or conditions of this Agreement and does not cure such failure in a reasonable period of time after becoming aware of such noncompliance. If all Recipient's rights under this Agreement terminate, Recipient agrees to cease use and distribution of the Program as soon as reasonably practicable. However, Recipient's obligations under this Agreement and any licenses granted by Recipient relating to the Program shall continue and survive.
+
+Everyone is permitted to copy and distribute copies of this Agreement, but in order to avoid inconsistency the Agreement is copyrighted and may only be modified in the following manner. The Agreement Steward reserves the right to publish new versions (including revisions) of this Agreement from time to time. No one other than the Agreement Steward has the right to modify this Agreement. The Eclipse Foundation is the initial Agreement Steward. The Eclipse Foundation may assign the responsibility to serve as the Agreement Steward to a suitable separate entity. Each new version of the Agreement will be given a distinguishing version number. The Program (including Contributions) may always be distributed subject to the version of the Agreement under which it was received. In addition, after a new version of the Agreement is published, Contributor may elect to distribute the Program (including its Contributions) under the new version. Except as expressly stated in Sections 2(a) and 2(b) above, Recipient receives no rights or licenses to the intellectual property of any Contributor under this Agreement, whether expressly, by implication, estoppel or otherwise. All rights in the Program not expressly granted under this Agreement are reserved.
+
+This Agreement is governed by the laws of the State of New York and the intellectual property laws of the United States of America. No party to this Agreement will bring a legal action under this Agreement more than one year after the cause of action arose. Each party waives its rights to a jury trial in any resulting litigation.
+================================================================
+
 github.com/elazarl/go-bindata-assetfs
 https://github.com/elazarl/go-bindata-assetfs
 ----------------------------------------------------------------
@@ -17600,6 +17692,33 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 ================================================================
 
+github.com/secure-io/sio-go
+https://github.com/secure-io/sio-go
+----------------------------------------------------------------
+MIT License
+
+Copyright (c) 2019 SecureIO
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+================================================================
+
 github.com/shirou/gopsutil
 https://github.com/shirou/gopsutil
 ----------------------------------------------------------------
@@ -19115,6 +19234,39 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 golang.org/x/time
 https://golang.org/x/time
+----------------------------------------------------------------
+Copyright (c) 2009 The Go Authors. All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+
+   * Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+   * Redistributions in binary form must reproduce the above
+copyright notice, this list of conditions and the following disclaimer
+in the documentation and/or other materials provided with the
+distribution.
+   * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+================================================================
+
+golang.org/x/tools
+https://golang.org/x/tools
 ----------------------------------------------------------------
 Copyright (c) 2009 The Go Authors. All rights reserved.
 

--- a/cmd/test-utils_test.go
+++ b/cmd/test-utils_test.go
@@ -77,6 +77,8 @@ func init() {
 	// Set system resources to maximum.
 	setMaxResources()
 
+	logger.Disable = true
+
 	// Uncomment the following line to see trace logs during unit tests.
 	// logger.AddTarget(console.New())
 }
@@ -536,8 +538,6 @@ func newTestConfig(bucketLocation string, obj ObjectLayer) (err error) {
 	if err = newSrvConfig(obj); err != nil {
 		return err
 	}
-
-	globalServerConfig.Logger.Console.Enabled = false
 
 	// Set a default region.
 	globalServerConfig.SetRegion(bucketLocation)

--- a/go.mod
+++ b/go.mod
@@ -17,7 +17,7 @@ require (
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/djherbis/atime v1.0.0
 	github.com/dustin/go-humanize v1.0.0
-	github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
+	github.com/eclipse/paho.mqtt.golang v1.2.0
 	github.com/elazarl/go-bindata-assetfs v1.0.0
 	github.com/fatih/color v1.7.0
 	github.com/fatih/structs v1.1.0

--- a/pkg/event/target/mqtt.go
+++ b/pkg/event/target/mqtt.go
@@ -1,5 +1,5 @@
 /*
- * MinIO Cloud Storage, (C) 2018 MinIO, Inc.
+ * MinIO Cloud Storage, (C) 2018-2019 MinIO, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -88,6 +88,7 @@ type MQTTTarget struct {
 	args       MQTTArgs
 	client     mqtt.Client
 	store      Store
+	quitCh     chan struct{}
 	loggerOnce func(ctx context.Context, err error, id interface{}, kind ...interface{})
 }
 
@@ -110,15 +111,15 @@ func (target *MQTTTarget) send(eventData event.Event) error {
 	}
 
 	token := target.client.Publish(target.args.Topic, target.args.QoS, false, string(data))
-	token.Wait()
-	if token.Error() != nil {
-		return token.Error()
+	if !token.WaitTimeout(reconnectInterval * time.Second) {
+		return errNotConnected
 	}
-	return nil
+	return token.Error()
 }
 
 // Send - reads an event from store and sends it to MQTT.
 func (target *MQTTTarget) Send(eventKey string) error {
+	// Do not send if the connection is not active.
 	if !target.client.IsConnectionOpen() {
 		return errNotConnected
 	}
@@ -158,11 +159,19 @@ func (target *MQTTTarget) Save(eventData event.Event) error {
 
 // Close - does nothing and available for interface compatibility.
 func (target *MQTTTarget) Close() error {
+	target.client.Disconnect(100)
+	close(target.quitCh)
 	return nil
 }
 
 // NewMQTTTarget - creates new MQTT target.
 func NewMQTTTarget(id string, args MQTTArgs, doneCh <-chan struct{}, loggerOnce func(ctx context.Context, err error, id interface{}, kind ...interface{})) (*MQTTTarget, error) {
+	if args.MaxReconnectInterval == 0 {
+		// Default interval
+		// https://github.com/eclipse/paho.mqtt.golang/blob/master/options.go#L115
+		args.MaxReconnectInterval = 10 * time.Minute
+	}
+
 	options := mqtt.NewClientOptions().
 		SetClientID("").
 		SetCleanSession(true).
@@ -175,36 +184,38 @@ func NewMQTTTarget(id string, args MQTTArgs, doneCh <-chan struct{}, loggerOnce 
 
 	client := mqtt.NewClient(options)
 
-	// The client should establish a first time connection.
-	// Connect() should be successful atleast once to publish events.
-	token := client.Connect()
-
 	target := &MQTTTarget{
 		id:         event.TargetID{ID: id, Name: "mqtt"},
 		args:       args,
 		client:     client,
+		quitCh:     make(chan struct{}),
 		loggerOnce: loggerOnce,
 	}
 
-	// Retries until the clientID gets registered.
+	token := client.Connect()
 	retryRegister := func() {
-		// Repeat the pings until the client registers the clientId and receives a token.
 		for {
-			var terr error
+		retry:
 			select {
 			case <-doneCh:
 				return
+			case <-target.quitCh:
+				return
 			default:
-				terr = token.Error()
-				if token.Wait() && terr == nil {
-					// Connected
+				ok := token.WaitTimeout(reconnectInterval * time.Second)
+				if ok && token.Error() != nil {
+					target.loggerOnce(context.Background(),
+						fmt.Errorf("Previous connect failed with %s attempting a reconnect",
+							token.Error()),
+						target.ID())
+					time.Sleep(reconnectInterval * time.Second)
+					token = client.Connect()
+					goto retry
+				}
+				if ok {
+					// Successfully connected.
 					return
 				}
-				// Reconnecting
-				time.Sleep(reconnectInterval * time.Second)
-				terr = fmt.Errorf("Previous connect failed with %s, attempting a reconnect", terr)
-				target.loggerOnce(context.Background(), terr, target.ID())
-				token = client.Connect()
 			}
 		}
 	}

